helm/hpcc/templates/_helpers.tpl

```diff
@@ -214,15 +214,25 @@ storage:
 {{- range $plane := $planes }}
  {{- if not $plane.disabled }}
   - name: {{ $plane.name | quote }}
-  {{- $planeYaml := omit $plane "name" "pvc" "storageClass" "storageSize" "subPath" -}}
+  {{- $planeYaml := omit $plane "name" "pvc" "storageClass" "storageSize" "subPath" "numMounts" -}}
   {{- if $plane.subPath -}}
    {{- $_ := set $planeYaml "prefix" (printf "%s/%s" $planeYaml.prefix $plane.subPath) -}}
   {{- end -}}
   {{- if and (eq "data" $plane.category) (not $plane.defaultSprayParts) -}}
    {{- $_ := set $planeYaml "defaultSprayParts" (include "hpcc.getMaxNumWorkers" $ | int) -}}
   {{- end -}}
+
+  {{- /* Remove pvc-related properties from the aliases*/ -}}
+  {{- if $plane.aliases }}
+   {{- $_ := set $planeYaml "aliases" (deepCopy $plane.aliases) -}}
```

Comment:
not sure this is necessary? "aliases" will already be a key of $planeYaml afaics.

Reply:
$planeYaml is already a copy of the original dictionary though, because 'omit' which created it, creates a new dictionary:
"The omit function is similar to pick, except it returns a new dict with all the keys that do not match the given keys."

.. which is desirable, because of the other preceding set $planeXML ... calls, that alter the copy and not the original.

I assumed that te copy (via omit) also included a copy of it's child dictionaries (like "aliases") too ?

Reply:
$planeYaml is a copy of the dictionary - but I don't think the items in the dictionary are copies.  So if you modify them they will modify both.

Reply:
This code was actually wrong, and revealed exactly this problem when I added the consistency check for the number of devices!